### PR TITLE
Scheduler - All Day Panel works incorrectly with certain appointments (T1177505)

### PR DESCRIPTION
**Pull request by pomahtri**

### CODE REVIEW

File: packages/devextreme/testing/testcafe/tests/scheduler/appointments/allDay/allDay.ts [hunk 1]
```diff
@@ -86,3 +86,53 @@ test('it should skip weekend days in timelineWorkWeek', async (t) => {
...
+
+test('should work correctly for unsorted dataSource', async (t) => {
+  const {
+    takeScreenshot,
+    compareResults,
+  } = createScreenshotsComparer(t);
+
+  await t
+    .expect(await takeScreenshot('allDay-unsorted-datasource.png'))
+    .ok();
```
Comment (williamvinogradov) [inline packages/devextreme/testing/testcafe/tests/scheduler/appointments/allDay/allDay.ts]: Let's screenshot only a workspace, e.g:

```typescript
const SCHEDULER_SELECTOR = '#container';

test('...', async (t) => {
  const scheduler = new Scheduler(SCHEDULER_SELECTOR);
  
  // other test code.
  
  await takeScreenshot('allDay-unsorted-datasource.png', scheduler.workSpace);
  
  // other test code.
});
```

It helps us don't update this etalon when something outside the scheduler's workspace will change.

File: packages/devextreme/testing/testcafe/tests/scheduler/appointments/allDay/allDay.ts [hunk 2]
```diff
@@ -86,3 +86,53 @@ test('it should skip weekend days in timelineWorkWeek', async (t) => {
...
+    .expect(compareResults.isValid())
+    .ok(compareResults.errorMessages());
+}).before(async () => {
+  await createWidget(
+    'dxScheduler',
+    {
+      dataSource: [{
+        id: 2,
+        text: '2',
+        startDate: new Date('2020-11-26T10:30:00.000'),
```
Comment (williamvinogradov) [inline packages/devextreme/testing/testcafe/tests/scheduler/appointments/allDay/allDay.ts]: Let's change this appointment start date to see it on the screenshot:
E.g. to ```2020-11-26T02:30:00.000```

File: packages/devextreme/testing/testcafe/tests/scheduler/appointments/allDay/allDay.ts [hunk 3]
```diff
@@ -86,3 +86,53 @@ test('it should skip weekend days in timelineWorkWeek', async (t) => {
...
+      }, {
+        id: 5,
+        text: '5',
+        startDate: new Date('2020-11-27T00:00:00.000'),
+        endDate: new Date('2020-11-27T00:00:00.000'),
+        allDay: true,
+      }, {
+        id: 1,
+        text: '1',
+        startDate: new Date('2020-11-25T22:20:00.000'),
```
Comment (williamvinogradov) [inline packages/devextreme/testing/testcafe/tests/scheduler/appointments/allDay/allDay.ts:125]: Not a review comment.

You can set ```startDate``` and ```endDate``` by date string without using a js Date
[Link to the example](https://codepen.io/williamvinogradov/pen/poQOQWE)

File: packages/devextreme/testing/testcafe/tests/scheduler/appointments/allDay/allDay.ts (hunk 2 above)
Comment (pomahtri) [reply packages/devextreme/testing/testcafe/tests/scheduler/appointments/allDay/allDay.ts]: reproduced without this appointment, so I've deleted it

**Comment by williamvinogradov** [comment]
It looks like we forgot to change the etalon after testcafe test case fix
That's why the CI is failing on the scheduler checks.

File: packages/devextreme/js/ui/scheduler/appointments/rendering_strategies/strategy_vertical.js [hunk 4]
```diff
@@ -362,7 +372,7 @@ class VerticalRenderingStrategy extends BaseAppointmentsStrategy {
     }
 
     _sortCondition(a, b) {
```
Comment (williamvinogradov) [inline packages/devextreme/js/ui/scheduler/appointments/rendering_strategies/strategy_vertical.js:364]: Let's rewrite it more obvious way, e.g:

```javascript
_sortCondition(a,b) {
    if(a.allDay === b.allDay) {
        const isBothAllDay = a.allDay && b.allDay;
        return this.groupOrientation === 'vertical'
        && isBothAllDay
            ? this._columnCondition(a, b)
            : this._rowCondition(a, b);
    }

    return a.allDay ? 1 : -1;
}
```

**Review by williamvinogradov — APPROVED** [review]

**Pull request merged**